app/models/response.rb

```diff
@@ -80,6 +89,17 @@ def self.between_dates(from, to)
     )
   end
 
+  def self.find_by_identifier(identifier, token)
+    person = Person.find_by_external_identifier(identifier)
+    return nil unless person
+
+    responses = person.protocol_subscriptions&.active&.map { |sub| sub.responses&.invited }.flatten
+    return nil if responses.blank?
+
+    responses.each { |resp| return resp if resp.invitation_token&.token == token }
```

Comment:
Is dit efficienter dan simpelweg:
```ruby
 InvitationToken.where(token: token).each do |invitation_token|
   next unless invitation_token.response.protocol_subscription.person.external_identifier
   return invitation_token.response
 end
```
?

Ik weet niet precies of je next en return zo in een each block mag gebruiken maar je snapt t idee. Volgens mij kun je gemiddeld genomen zo een behoorlijk aantal queries sparen.

Reply:
yeah mee eens, het punt is dat dat Bcrypt een hash oplevert die je niet op deze manier kan checken. Ze overloaden die == functie en doen wat magie daar. We zouden nog een keer in die bcrypt gem kunnen duiken en kijken of we hier zelf iets voor kunnen maken.